crates/iota-json-rpc-types/src/iota_transaction.rs

```diff
@@ -2049,6 +2049,49 @@ impl From<Argument> for IotaArgument {
     }
 }
 
+/// An argument to a transaction in a programmable transaction block
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema, PartialEq, Eq)]
+pub enum IotaArgumentV2 {
```

Comment:
The premise here is that It is ok to break the Rust API as long as we are backward compatible on the web interface. For example this is satisfied by the addition of the new `RPCTransactionRequestParam`.

On that basis we could think of adding the following type instead 
```rust
#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema, PartialEq, Eq)]
#[serde(untagged)]
pub enum PtbInput {
     CallArg(IotaJsonValue)
     PtbRef(IotaArgument)
}
```
This is backward compatible with `IotaJsonValue` with respect to deserialization, and could be used in `MoveCallParams` directly. So without the need for a `MoveCallParamsV2`. And it would not require `v2` methods in the `transaction-builder` either.

What do you think?

Reply:
Would definitely clean some things up, but would be a breaking change in the transaction builder crate, so not sure if we want that
Draft PR with `PtbInput` change https://github.com/iotaledger/iota/pull/8779
And total diff with all changes:
https://github.com/iotaledger/iota/compare/develop...dev-tools/PtbInput

Reply:
Then you can limit the introduction of new methods only in the transaction builder. With a `single_move_call_with_ptb_inputs` method (similar to the `_v2` variant).

Reply:
Thanks for the suggestion, I made the change and like it much more now!